src/cart/crt_corpc.c

```diff
@@ -968,10 +976,11 @@ crt_corpc_req_hdlr(struct crt_rpc_priv *rpc_priv)
 		crt_corpc_reply_hdlr(&cb_info);
 	} else {
 		rc = crt_rpc_common_hdlr(rpc_priv);
-		if (rc != 0)
+		if (rc != 0) {
 			RPC_ERROR(rpc_priv,
-				  "crt_rpc_common_hdlr failed, rc: %d\n",
-				  rc);
+				  "crt_rpc_common_hdlr failed, rc: %d\n", rc);
+			crt_corpc_fail_parent_rpc(rpc_priv, rc);
```

Comment:
the err handling is a kind of complex, but seems this change possibly cause two reply msg send to the parent, one is here it returns error then crt_corpc_common_hdlr returns error and trigger crt_hg_reply_error_send.
Another one is the child RPC is still inflight and when all children's reply arrive it possibly trigger the aggregation done and send another reply to parent. this possibly cause unexpected result.

seems a simpler fix is here we can call "crt_corpc_fail_child_rpc(rpc_priv, 1, rc)", and reset rc = 0.  thoughts?

Reply:
After reading more of corpc code, this sounds like it should do the trick with the single response, but it still will require addref before forwarding rpc to children (with corresponding decref in corpc reply handler). I've uploaded a new patch

Reply:
With a local test that emulates failure of corpc, with this fix we still are leaking an rpc handle on the node that has a failure. Continuing to investigate...